webhooks-extension/config/triggers_prototype/ingress.yaml

```diff
@@ -0,0 +1,109 @@
+apiVersion: tekton.dev/v1alpha1
+kind: Task
+metadata:
+  name: ingress-task
+  namespace: tekton-pipelines
+spec:
+  inputs:
+    params:
+    - name: Mode
+      description: Whether to create or delete Ingress
+    - name: CreateCertificate
+      description: Whether to create a certificate
+      default: "true"
+    - name: CertificateKeyPassphrase
+      description: The phrase that protects private key
+      default: "phrase1"
+    - name: CertificateSecretName
+      description: The secret name for the Ingress certificate
+      default: "secret1"
+    - name: ExternalUrl 
+      description: The external access URL
+      default: "https://$(inputs.params.EventListenerName).replaceme.nip.io"
+    - name: EventListenerName 
+      description: The EventListener name
+      default: "listener"
+    - name: Namespace
+      description: "The namespace to use for where the secret will be created/deleted"
+      default: "tekton-pipelines"
+  steps:
+  - name: generate-certificate
+    image: frapsoft/openssl
```

Comment:
Probably need to check licenses on the images being used

Reply:
Pulled and ran interactively with
`docker run -it --entrypoint /bin/ash frapsoft/openssl`

No license file found in here
openssl binary itself, which we know is in here (obv) is version 1.0.2 from September 2016 (I hate how old this image is, but not enough that I want us to maintain our own openssl image)

```
For any release made before OpenSSL 3.0.0 (namely the 1.1.1, 1.1.0, 1.0.2, and all prior releases including those not currently supported), the dual OpenSSL and SSLeay license applies. Note that this is also true for any updates to those releases -- the "letter suffix" -- no matter when they are made. It also applies to the git branches for all those releases, and to any public forks that have not rebased to master (or 3.0.0).
```

Also, I remember doing export regulation training, openssl is definitely something you can use for encryption. Sooo we could recommend running a script we provide, assuming they've got openssl, then mount in as a secret, and have that certificate used by this task.

Reply:
Alternative approach which I lovingly refer to as BYOC at https://github.com/tektoncd/experimental/pull/276